synapse/handlers/room.py

```diff
@@ -1459,14 +1458,9 @@ async def get_event_for_timestamp(
                 timestamp,
             )
 
-            # Find other homeservers from the given state in the room
-            curr_state = await self._storage_controllers.state.get_current_state(
-                room_id
+            likely_domains = (
+                await self._storage_controllers.state.get_current_hosts_in_room(room_id)
             )
-            curr_domains = get_domains_from_state(curr_state)
-            likely_domains = [
-                domain for domain, depth in curr_domains if domain != self.server_name
-            ]
 
             # Loop through each homeserver candidate until we get a succesful response
             for domain in likely_domains:
```

Comment:
I think these CI failures are because we're just querying our own server when we want some other federated one. `get_current_hosts_in_room` doesn't exclude our own server.

 - https://github.com/matrix-org/synapse/runs/7929900798?check_suite_focus=true#step:4:3822
 - https://github.com/matrix-org/synapse/runs/7929900765?check_suite_focus=true#step:4:3780